test/e2e/network_segmentation_services.go

```diff
@@ -433,3 +459,136 @@ func checkConnectionOrNoConnectionToNodePort(f *framework.Framework, clientPod *
 		framework.ExpectNoError(err, fmt.Sprintf("Failed to verify that %s", msg))
 	}
 }
+
+func checkConnectionToExternalIPs(f *framework.Framework, clientPod *v1.Pod, service *v1.Service, expectedOutput string) {
+	checkConnectionOrNoConnectionToExternalIPs(f, clientPod, service, expectedOutput, true)
+}
+
+func checkNoConnectionToExternalIPs(f *framework.Framework, clientPod *v1.Pod, service *v1.Service) {
+	checkConnectionOrNoConnectionToExternalIPs(f, clientPod, service, "", false)
+}
+
+func checkConnectionOrNoConnectionToExternalIPs(f *framework.Framework, clientPod *v1.Pod, service *v1.Service, expectedOutput string, shouldConnect bool) {
```

Comment:
this `shouldConnect` boolean is super ugly. 

can't you refactor this to check in the caller if the connection should happen or not ? Like return an error, and match it in the caller - `Should(Succeed())` vs `MatchError(<blabla>)`

Reply:
This is done like the other checks, `checkConnectionOrNoConnectionToClusterIPs`, Don't want to refactor the whole test suite here.